SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/ViewDeltaTest.cpp

```diff
@@ -230,6 +230,67 @@ VIEWDELTA_TEST(GIVEN_entity_and_auth_component_in_view_WHEN_authority_lost_THEN_
 	return true;
 }
 
+VIEWDELTA_TEST(GIVEN_entity_and_component_in_view_WHEN_authority_gained_with_no_component_data_THEN_component_removed_from_view)
+{
+	ViewDelta InputDelta;
+	EntityView InputView;
+	FComponentSetData ComponentSetData = GetTestComponentSetData();
+	ComponentData TestComponentData = CreateTestComponentData(TestComponentId, TestComponentValue);
+
+	AddEntityToView(InputView, TestEntityId);
+	AddComponentToView(InputView, TestEntityId, TestComponentData.DeepCopy());
+
+	EntityComponentOpListBuilder OpListBuilder;
+	// Set authority with no component data - implying the component should be removed.
+	OpListBuilder.SetAuthority(TestEntityId, TestComponentSetId, WORKER_AUTHORITY_AUTHORITATIVE, {});
+	SetFromOpList(InputDelta, InputView, MoveTemp(OpListBuilder), ComponentSetData);
+
+	EntityView ExpectedView;
+	AddEntityToView(ExpectedView, TestEntityId);
+	AddAuthorityToView(ExpectedView, TestEntityId, TestComponentSetId);
+
+	ExpectedViewDelta ExpectedDelta;
+	ExpectedDelta.AddEntityDelta(TestEntityId, ExpectedViewDelta::UPDATE);
+	ExpectedDelta.AddComponentRemoved(TestEntityId, TestComponentId);
+	ExpectedDelta.AddAuthorityGained(TestEntityId, TestComponentSetId);
+
+	TestTrue("View Deltas are equal", ExpectedDelta.Compare(InputDelta));
+	TestTrue("Views are equal", CompareViews(InputView, ExpectedView));
+
+	return true;
+}
+
+VIEWDELTA_TEST(GIVEN_entity_with_no_components_WHEN_authority_gained_with_non_empty_component_data_THEN_component_added_to_view)
+{
+	ViewDelta InputDelta;
+	EntityView InputView;
+	FComponentSetData ComponentSetData = GetTestComponentSetData();
+	ComponentData TestComponentData = CreateTestComponentData(TestComponentId, TestComponentValue);
```

Comment:
Worth using a component set with more than a single component id so we can
1/ validate multiple components can get added
2/ there's not confusion between component set id and component id (as they're currently always the same in our tests)

Reply:
Good points. Added.